pandas/tests/indexes/multi/test_indexing.py

```diff
@@ -437,3 +437,21 @@ def test_timestamp_multiindex_indexer():
     )
     should_be = pd.Series(data=np.arange(24, len(qidx) + 24), index=qidx, name="foo")
     tm.assert_series_equal(result, should_be)
+
+
+def test_multiindex_loc_order():
+    # GH 22797
+    # Try to respect order of keys given for MultiIndex.loc
+    df = pd.DataFrame(
+        np.arange(12).reshape((4, 3)),
+        index=[["a", "a", "b", "b"], [1, 2, 1, 2]],
+        columns=[["Ohio", "Ohio", "Colorado"], ["Green", "Red", "Green"]],
+    )
+
+    res = df.loc[["b", "a"], :]
```

Comment:
Can you test selection of columns as well? Looks like you have the data for it already

Reply:
Added a test for `df.loc[:, ["Colorado", "Ohio"]]` and  df.loc[:, (["Colorado", "Ohio"], ["Red", "Green"])]`.

Also, I add this`df.loc[(slice(None), [2, 1]), :]`,  testing if the result was
```
     Ohio     Colorado
    Green Red    Green
a 2     3   4        5
b 2     9  10       11
a 1     0   1        2
b 1     6   7        8
```

but maybe a preferable result is this
```
     Ohio     Colorado
    Green Red    Green
a 2     3   4        5
  1     0   1        2
b 2     9  10       11
  1     6   7        8
```
This would be harder to obtain given the current implementation.